packages/@aws-cdk/aws-appmesh/lib/route-spec.ts

```diff
@@ -34,6 +34,255 @@ export interface HttpRouteMatch {
    * and you want the route to match requests to my-service.local/metrics, your prefix should be /metrics.
    */
   readonly prefixPath: string;
+
+  /**
+   * Specifies the client request headers to match on. All specified headers
+   * must match for the route to match.
+   *
+   * @default - do not match on headers
+   */
+  readonly headers?: HeaderMatch[];
+
+  /**
+   * The HTTP client request method to match on.
+   *
+   * @default - do not match on request method
+   */
+  readonly method?: HttpRouteMatchMethod;
+
+  /**
+   * The client request protocol to match on. Applicable only for HTTP2 routes.
+   *
+   * @default - do not match on HTTP2 request protocol
+   */
+  readonly protocol?: HttpRouteProtocol;
+}
+
+/**
+ * Supported values for matching routes based on the HTTP request method
+ */
+export enum HttpRouteMatchMethod {
+  /**
+   * GET request
+   */
+  GET = 'GET',
+
+  /**
+   * HEAD request
+   */
+  HEAD = 'HEAD',
+
+  /**
+   * POST request
+   */
+  POST = 'POST',
+
+  /**
+   * PUT request
+   */
+  PUT = 'PUT',
+
+  /**
+   * DELETE request
+   */
+  DELETE = 'DELETE',
+
+  /**
+   * CONNECT request
+   */
+  CONNECT = 'CONNECT',
+
+  /**
+   * OPTIONS request
+   */
+  OPTIONS = 'OPTIONS',
+
+  /**
+   * TRACE request
+   */
+  TRACE = 'TRACE',
+
+  /**
+   * PATCH request
+   */
+  PATCH = 'PATCH',
+}
+
+/**
+ * Supported :scheme options for HTTP2
+ */
+export enum HttpRouteProtocol {
+  /**
+   * Match HTTP requests
+   */
+  HTTP = 'http',
+
+  /**
+   * Match HTTPS requests
+   */
+  HTTPS = 'https',
+}
+
+/**
+ * Configuration for `HeaderMatch`
+ */
+export interface HeaderMatchConfig {
+  /**
+   * The HTTP route header.
+   */
+  readonly httpRouteHeader: CfnRoute.HttpRouteHeaderProperty;
+}
+
+/**
+ * Used to generate header matching methods.
+ */
+export abstract class HeaderMatch {
+  /**
+   * The value of the header with the given name in the request must match the
+   * specified value exactly.
+   *
+   * @param headerName Test against the value of this HTTP request header
```

Comment:
How about just

```suggestion
   * @param headerName the name of the HTTP header to match against
```

? 🙂

Same comment everywhere below.